substrate/bin/utils/chain-spec-builder/bin/main.rs

```diff
@@ -71,19 +83,45 @@ fn main() -> Result<(), String> {
 			let sudo_account =
 				chain_spec::get_account_id_from_seed::<sr25519::Public>(&sudo_seed).to_ss58check();
 
-			(authority_seeds, nominator_accounts, endowed_accounts, sudo_account)
+			generate_chain_spec(authority_seeds, nominator_accounts, endowed_accounts, sudo_account)
 		},
-		ChainSpecBuilder::New {
+		ChainSpecBuilderCmd::New(NewCmd {
 			authority_seeds,
 			nominator_accounts,
 			endowed_accounts,
 			sudo_account,
-			..
-		} => (authority_seeds, nominator_accounts, endowed_accounts, sudo_account),
-	};
+		}) =>
+			generate_chain_spec(authority_seeds, nominator_accounts, endowed_accounts, sudo_account),
+		ChainSpecBuilderCmd::Runtime(cmd) => generate_chain_spec_for_runtime(&cmd),
+		ChainSpecBuilderCmd::Edit(EditCmd {
+			ref input_chain_spec,
+			ref runtime_wasm_path,
+			convert_to_raw,
+		}) => {
+			let mut chain_spec = GenericChainSpec::<()>::from_json_file(input_chain_spec.clone())?;
+			if let Some(path) = runtime_wasm_path {
+				chain_spec
+					.set_code(&fs::read(path.as_path()).expect("wasm blob file is readable")[..])
+					.into()
+			}
 
-	let json =
-		generate_chain_spec(authority_seeds, nominator_accounts, endowed_accounts, sudo_account)?;
+			chain_spec.as_json(convert_to_raw)
+		},
+		ChainSpecBuilderCmd::Verify(VerifyCmd { ref input_chain_spec, ref runtime_wasm_path }) => {
+			write_chain_spec = false;
+			let mut chain_spec = GenericChainSpec::<()>::from_json_file(input_chain_spec.clone())?;
+			if let Some(path) = runtime_wasm_path {
+				chain_spec
+					.set_code(&fs::read(path.as_path()).expect("wasm blob file is readable")[..])
```

Comment:
How do you know that the file is readable? You should return an error to the user.